pkg/component/watch.go

```diff
@@ -109,22 +109,37 @@ func addRecursiveWatch(watcher *fsnotify.Watcher, path string, ignores []string)
 
 var UserRequestedWatchExit = fmt.Errorf("safely exiting from filesystem watch based on user request")
 
+// WatchParameters is designed to hold the controllables and attributes that the watch function works on
+type WatchParameters struct {
```

Comment:
Please move type definitions to the beginning of the file

Reply:
My bad :disappointed: 
Fixed now